src/memory/AmdFlash.hh

```diff
@@ -227,12 +228,25 @@ public:
 
 	enum class State { IDLE, IDENT, CFI, STATUS, PRGERR };
 
+	struct Sector {
+		const size_t address = 0;
+		const power_of_two<size_t> size = 1;
+		const bool writeProtect = false;
+		ptrdiff_t writeAddress = -1;
+		const uint8_t* readAddress = nullptr;
+
+		std::weak_ordering operator<=>(const Sector& sector) const { return address <=> sector.address; }
+		bool operator==(const Sector& sector) const = default;
+	};
+
 private:
 	AmdFlash(const std::string& name, const ValidatedChip& chip,
 	         const Rom* rom, std::span<const bool> writeProtectSectors,
 	         const DeviceConfig& config);
-	struct GetSectorInfoResult { size_t sector, sectorSize, offset; };
-	[[nodiscard]] GetSectorInfoResult getSectorInfo(size_t address) const;
+
+	[[nodiscard]] size_t getSectorIndex(size_t address) const;
+	[[nodiscard]] Sector& getSector(size_t address) { return sectors[getSectorIndex(address)]; };
```

Comment:
The non-const version of `getSector()` isn't needed ATM (I checked). And it also shouldn't be needed in the future (I think).

Reply:
It will be needed, because I’m going to add modifyable members, such as:

* `erasePending` To track which sectors are marked for erase. It is possible to issue multiple sector erase commands in one go (under a time-out). Currently working on that as well as implementing proper timing in general.
* `modified` To track which sectors have been modified from the original. On the backburner a bit, but will come back to that soon.